frontend/src/app/components/enterprise/enterprise-modal/enterprise-trial.modal.ts

```diff
@@ -0,0 +1,118 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See docs/COPYRIGHT.rdoc for more details.
+// ++
+
+import {AfterViewInit, ChangeDetectorRef, Component, ElementRef, Inject, Input, ViewChild} from "@angular/core";
+import {FormControl, FormGroup} from "@angular/forms";
+import {OpModalComponent} from "app/components/op-modals/op-modal.component";
+import {OpModalLocalsToken} from "app/components/op-modals/op-modal.service";
+import {OpModalLocalsMap} from "app/components/op-modals/op-modal.types";
+import {I18nService} from "app/modules/common/i18n/i18n.service";
+import {EETrialFormComponent} from "core-components/enterprise/enterprise-modal/enterprise-trial-form/ee-trial-form.component";
+import {EnterpriseTrialService} from "core-components/enterprise/enterprise-trial.service";
+
+export interface EnterpriseTrialOptions {
+  closeByEscape?:boolean;
+  showClose?:boolean;
+  closeByDocument?:boolean;
+}
+
+@Component({
+  selector: 'enterprise-trial-modal',
+  templateUrl: './enterprise-trial.modal.html',
+  styleUrls: ['./enterprise-trial.modal.sass']
+})
+export class EnterpriseTrialModal extends OpModalComponent implements AfterViewInit {
+  @ViewChild(EETrialFormComponent, { static: false }) formComponent:EETrialFormComponent;
+  @Input() public opReferrer:string;
+
+  public trialForm:FormGroup;
+
+  public showClose:boolean;
+  public confirmed:boolean;
+  public cancelled = false;
+  public status:string;
+  public errorMsg:string|undefined;
+
+  private options:EnterpriseTrialOptions;
+
+  public text = {
+    button_submit: this.I18n.t('js.modals.button_submit'),
+    button_cancel: this.I18n.t('js.modals.button_cancel'),
+    button_continue: this.I18n.t('js.button_continue'),
+    close_popup: this.I18n.t('js.close_popup_title'),
+    heading_confirmation: this.I18n.t('js.admin.enterprise.trial.confirmation'),
+    heading_next_steps: this.I18n.t('js.admin.enterprise.trial.next_steps'),
+    heading_test_ee: this.I18n.t('js.admin.enterprise.trial.test_ee'),
+    next_step: this.I18n.t('js.admin.enterprise.trial.next_step'),
+    quick_overview: this.I18n.t('js.admin.enterprise.trial.quick_overview')
+  };
+
+  constructor(readonly elementRef:ElementRef,
+              @Inject(OpModalLocalsToken) public locals:OpModalLocalsMap,
+              readonly cdRef:ChangeDetectorRef,
+              readonly I18n:I18nService,
+              public eeTrialService:EnterpriseTrialService) {
+    super(locals, cdRef, elementRef);
+
+    // modal configuration
+    this.options = locals.options || {};
+    this.closeOnEscape = _.defaultTo(this.options.closeByEscape, false);
+    this.closeOnOutsideClick = _.defaultTo(this.options.closeByDocument, true);
```

Comment:
Why do you set these variables so complicated? `locals.options` is not set when you call the modal, so all of this is somehow obsolete. You could also set `this.closeOnOutsideClick = true;`. The interface is then also not needed.